### ISSUE TITLE
Feature: Turn on LEDs only when actively printing

### ISSUE THREAD

**Issue by mariusstrom** [DutchDevelop/BLLEDController#2]
It would be amazing if there was a setting in BLLED to only have the LEDs triggered when the printer is not idling. For example, lit white when printing, red when errored, and green when complete. Otherwise, turn the LEDs off.

**Comment by bolsoncerrado** [comment]
How do you go from Green after completion to LEDs off if not manually?

**Comment by HotWheelz83** [comment]
I'd suggest using a door sensor and time out combo, so say leds turn green when the print finishes and then turn off after a user adjustable period (for me, I'd like 12 hours) or when the printer door is opened, whichever comes first.

**Comment by bolsoncerrado** [comment]
Now that you mention it, don't the X1 series come with door sensors already installed?

**Comment by TheDutchDev** [comment]
> Now that you mention it, don't the X1 series come with door sensors already installed?

I dont think so no, maybe the latest models? Mine atleast doesnt have it afaik. Bought it 3 or 4 months ago

**Comment by HotWheelz83** [comment]
Mine doesnt have a stock door sensor from factory, but it's trivial to add one on the outside of the door and frame.

**Comment by bolsoncerrado** [comment]
I though all of them had the sensors installed just not enabled in the firmware. If you look close to the protuberances in the clear Grey rubber when you open the door...

**Comment by bolsoncerrado** [comment]
There are even suggestions on their actual use:

https://github.com/bambulab/BambuStudio/issues/2751

**Comment by TheDutchDev** [comment]
Ah, my bad. @DutchDevelop also just told me there is a way to detect the door sensor.

That's actually pretty cool

**Comment by wile1411** [comment]
Some work on the code was recently completed by a community member 

The LEDs stay green after a print finishes. Once you open or close the door, the LEDs return to white
You can also use the door as a LED light switch to turn them off by closing the printer door twice in 6 seconds
Lastly, there is an setable inactivity timer. If it's not printing, the LEDs will turn off after that time expires.

Check the latest BLLC_19.3.24.bin version from official site https://www.dutchdevelop.com/bl-led-controller/
